features/welcome/components/AbstractWelcomePage.js

```diff
@@ -17,37 +22,42 @@ export class AbstractWelcomePageContainer extends Component {
 
         this.state = { roomName: '' };
 
-        this._onRoomNameChange = this._onRoomNameChange.bind(this);
         this._onJoinPress = this._onJoinPress.bind(this);
+        this._onRoomNameChange = this._onRoomNameChange.bind(this);
     }
 
     /**
-     * Handles 'change' event for the room name input field.
+     * Handles click on 'Join' button.
      *
-     * @param {string} value - Name for room.
      * @protected
      * @returns {void}
      */
-    _onRoomNameChange(value) {
-        this.setState({ roomName: value });
+    _onJoinPress() {
+        this.props.dispatch(navigate({
+            screen: APP_SCREEN.CONFERENCE,
+            room: this.state.roomName,
+            navigator: this.props.navigator
+        }));
     }
 
     /**
-     * Handles click on 'Join' button.
+     * Handles 'change' event for the room name input field.
      *
+     * @param {string} value - Name for room.
      * @protected
      * @returns {void}
      */
-    _onJoinPress() {
-        this.props.onJoin(this.state.roomName);
+    _onRoomNameChange(value) {
+        this.setState({ roomName: value });
     }
 }
 
 /**
- * WelcomePageContainer component's property types.
+ * WelcomePage component's property types.
```

Comment:
Well, AbstractWelcomePage is technically more accurate than WelcomePage. I know that given that WelcomePage is going to reuse them, one may claim that these are WelcomePage's property types. But there's no way for AbstractWelcomePage to know that.